src/hooks/sounds.tsx

```diff
@@ -35,33 +35,40 @@ export interface ExposedDataWithPlay extends ExposedData {
 
 export interface CustomSoundsState {
   soundConfigs: SoundConfigMap;
+  isLoaded: boolean;
 }
 
 export const initialValue: CustomSoundsState = {
   soundConfigs: {},
+  isLoaded: false,
 };
 
 export const CustomSoundsContext = createContext<CustomSoundsState>(
   initialValue
 );
 
 export const CustomSoundsProvider: React.FC = ({ children }) => {
+  const [isLoaded, setIsLoaded] = useState<boolean>(initialValue.isLoaded);
   const [soundConfigs, setSoundConfigs] = useState<SoundConfigMap>(
     initialValue.soundConfigs
   );
 
   // Fetch the sound configs data on first load
   useEffect(() => {
-    fetchSoundConfigs().then((soundConfigs) => {
-      setSoundConfigs(soundConfigs);
-    });
+    fetchSoundConfigs()
+      .then((soundConfigs) => {
+        setSoundConfigs(soundConfigs);
+        setIsLoaded(true);
+      })
+      .catch(() => setIsLoaded(true));
```

Comment:
Do we want to be setting `isLoaded` to `true` when we have an error? I guess this depends on how we intend to use this state.

If it's just meant to track that we 'currently are in the progress of loading data', then this makes sense (though would potentially be better set in a `.finally()` rather than in both the `.then()`/`.catch()`. If that's the case, I wonder if 'flipping' it to be `isLoading` would make more sense? Then we could derive `isLoaded` or similar from that if needs be.

If it's meant to represent that we successfully loaded the data from firestore, then this usage seems invalid.

Personally, I feel like having an indication that we successfully loaded the data from firestore seems like a useful thing to have here.

Reply:
In the case of an error, depending on what the error reason is, I wonder if there is something better we can do here to 'gracefully recover' (eg. perhaps retrying to fetch the data later or similar maybe? Not sure..)

Reply:
removed `isLoaded` try to find the better solution